src/main/java/org/mozilla/jss/netscape/security/x509/ExtensionsRequested.java

```diff
@@ -160,7 +160,7 @@ private void construct(DerValue dv) throws IOException {
         // now, the stream will be in the correct format
         stream.reset();
 
-        while (true) {
+        while (stream.available() != 0) {
```

Comment:
I'm not sure about this. According to [Javadoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#available()) the method:
> Returns an **estimate** of the number of bytes that can be read (or skipped over) from this input stream **without blocking**, which may be 0, or 0 when end of stream is detected.

I think `0` availability doesn't necessarily mean that it's the end of stream. It could just mean that there's no data that can be read now, but there might be more coming later, so we probably shouldn't terminate the loop based on this.

We probably need to do more investigation. Right now the `DerInputStream.getDerValue()` always returns a non-null object. It probably needs to return a `null` if it reaches the end of stream so we can terminate the loop based on that.

Reply:
> I think `0` availability doesn't necessarily mean that it's the end of stream. It could just mean that there's no data that can be read now, but there might be more coming later, so we probably shouldn't terminate the loop based on this.

Hmm, good point. I have reverted this part for now.